### PR TITLE
Test branch

### PR DESCRIPTION
**Pull request by santiagorodriguezbermudez**
# Test Branch Request

## In this branch we worked on:
- [x] Added RSpec methods to test:
- my_each
- my_each_with_index
- my_select
- my_any?
- my_all?
- my_none?
- my_count
- my_map
- my_inject

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    end
+
+    it 'My all checks correctly if the word length on all of its items is for a specific size' do
+      expect(array.my_all? { |word| word.length >= 5 }).to eql(false)
+    end
+
+    it 'All my elements contain a specific character' do
+      expect(array.my_all?(/t/)).to eql(false)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Test for when
- [x] when a class is passed as an argument.
- [x] when no block or argument is given

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    end
+
+    it 'My any will check if any of the items inside an array comply with the conditions' do
+      expect(array.my_any? { |word| word.length >= 5 }).to eql(false)
+    end
+
+    it 'Any of my elements contain a specific character' do
+      expect(array.my_any?(/t/)).to eql(true)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Test for when
- [x] when a class is passed as an argument.
- [x] when no block or argument is given

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    end
+
+    it 'My none will check if none of the items inside an array comply with the conditions' do
+      expect(array.my_none? { |word| word.length >= 5 }).to eql(true)
+    end
+
+    it 'None of my elements contain a specific character' do
+      expect(array.my_none?(/t/)).to eql(false)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Test for when
- [x] when a class is passed as an argument.
- [x] when no block or argument is given

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    end
+
+    it 'My map will return the concat result of the conditions given inside the block applied on each item' do
+      expect(array.my_map { |word| word + 'item' }).to eql(%w[countitem bearitem catitem fishitem lionitem])
+    end
+
+    it 'My map will return a new array with the results of the conditions given in the parameter applied on every item' do
+      expect(numbers.my_map(&:to_i)).to eql([1, 5, 8, 7, 9, 6, 5, 2, 3])
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Test for when
- [x] when no block or argument is given

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
 - [x] Lets edit this pull request, you are to use Gitflow for this project  
 - [x] The readme file should be descriptive, for example how to run test, how to clone this project should be included.  
 - [x] Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by santiagorodriguezbermudez** [comment]
## Code Review # 1

We made the following changes:
- [x] Rebase the pull request to merge into develop branch
- [x] Implemented new scenarios in my_any?, my_all?, my_none? and my_map? according to the TSE suggestions. 
- [x] Wrote Readme instructions to run tests.

**Comment by lekegitrepo** [comment]
#### Good job :clap: :clap:
#### Well done for implementing the required changes. This project is approved.

#### _Reviewed by Temitayo Adeleke_.

**Review by lekegitrepo — APPROVED** [review]
#### Status: Approved.

**Pull request merged**